gradle-plugin/src/main/groovy/io/realm/gradle/Realm.groovy

```diff
@@ -72,7 +72,11 @@ class Realm implements Plugin<Project> {
 
         project.android.registerTransform(new RealmTransformer(project))
 
-        project.repositories.add(project.getRepositories().jcenter())
+        if (project.repositories.isEmpty()) {
```

Comment:
❌ This only account for repositories added before the plugin is applied. If, for example, a user would have something like the following:
```
apply plugin: 'realm-android'

repositories {
    maven {
        url 'https://jcenter.my-company.com'
    }
}
```

There would be two repositories added where `jcenter.bintray.com` would still be the first one to resolve. A better solution, unfortunately using the dredded `afterEvaluate` would be something like this:
```
project.afterEvaluate {
    if (project.repositories.isEmpty()) {
        project.repositories.jcenter()
    }
}
```

This will react by waiting at the end of the evaluation of the script before checking the if no repositories are added. There are still issues when a user would also add repositories during an `afterEvaluate` bloc, but it should be quite limited. The Gradle team will hopefully provide an API to default repositories when none is specified without relying on `afterEvaluate` but I can't say when that will happen.